core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -914,11 +914,23 @@ class TransactionsTest extends IntegrationTestHarness {
 
     // Check that the epoch only increased by 1 when TV2 is disabled.
     // With TV2 and the latest EndTxnRequest version, the epoch will be bumped at the end of every transaction.
-    if (!isTV2Enabled) assertEquals((initialProducerEpoch + 1).toShort, producerStateEntry.producerEpoch)
+    if (!isTV2Enabled) {
+      producerStateEntry =
```

Comment:
If we duplicate this code, we should remove it from lines 911-913.
I am also taking a look at something.